src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java

```diff
@@ -560,6 +583,10 @@ public CreateContainerCmd fillContainerConfig(CreateContainerCmd containerConfig
             containerConfig.withUser(user);
         }
 
+        if (hostConfig != null && CollectionUtils.isNotEmpty(extraGroups)) {
+            hostConfig.withGroupAdd(extraGroups);
+        }
+
```

Comment:
The established pattern in this code (and in the docker-java code) is that "everyone calls getHostConfig() every time they need it" rather than calling it once and reusing it.
Please follow the existing pattern.

FYI there is some logic to this "extra code" - if there's a problem with the getHostConfig() method then, if we only ever call getHostConfig() when we need it, anyone who doesn't use the affected functionality will be able to continue working; if we always call getHostConfig() regardless of whether or not we need it _and_ there's a problem with it then everyone is screwed.
The principal of "retain existing functionality" is key here - we don't want all existing users to be affected by this new functionality unless they decide to use it.